crates/astria-sequencer/src/app/mod.rs

```diff
@@ -1000,6 +1014,16 @@ impl App {
             .await
             .context("failed executing transaction")?;
 
+        // flag mempool for cleaning if we ran a fee change action
+        for action in signed_tx.actions() {
+            match action {
+                Action::FeeAssetChange(_) | Action::FeeChange(_) => {
+                    self.recost_mempool = true;
+                }
+                _ => {}
+            }
+        }
```

Comment:
We can make this a little more efficient:
```suggestion
        self.recost_mempool = self.recost_mempool || signed_tx
            .actions()
            .iter()
            .any(|action| matches!(action, Action::FeeAssetChange(_) | Action::FeeChange(_)));
```
This way, if `self.recost_mempool` is already `true` before this statement, we won't iterate the actions.  Also, `any` is short-circuiting, so it stops iterating on the first instance where `matches!` returns `true`.